libs/providers/ofrep/src/lib/ofrep-provider.ts

```diff
@@ -0,0 +1,113 @@
+  }
+
+  public async resolveBooleanEvaluation(
+    flagKey: string,
+    defaultValue: boolean,
+    context: EvaluationContext,
+  ): Promise<ResolutionDetails<boolean>> {
+    const result = await this.ofrepApi.postEvaluateFlags(flagKey, { context });
+    return this.toResolutionDetails(result, defaultValue);
+  }
+
+  public async resolveStringEvaluation(
+    flagKey: string,
+    defaultValue: string,
+    context: EvaluationContext,
+  ): Promise<ResolutionDetails<string>> {
+    const result = await this.ofrepApi.postEvaluateFlags(flagKey, { context });
+    return this.toResolutionDetails(result, defaultValue);
+  }
+
+  public async resolveNumberEvaluation(
+    flagKey: string,
+    defaultValue: number,
+    context: EvaluationContext,
+  ): Promise<ResolutionDetails<number>> {
+    const result = await this.ofrepApi.postEvaluateFlags(flagKey, { context });
+    return this.toResolutionDetails(result, defaultValue);
+  }
+
+  public async resolveObjectEvaluation<U extends JsonValue>(
+    flagKey: string,
+    defaultValue: U,
+    context: EvaluationContext,
+  ): Promise<ResolutionDetails<U>> {
+    const result = await this.ofrepApi.postEvaluateFlags(flagKey, { context });
+    return this.toResolutionDetails(result, defaultValue);
+  }
+
+  private toResolutionDetails<T extends EvaluationFlagValue>(
+    result: OFREPApiEvaluationResult,
+    defaultValue: T,
+  ): ResolutionDetails<T> {
+    if (result.httpStatus !== 200) {
+      const code = result.value.errorCode;
+      const details = result.value.errorDetails;
+
+      switch (code) {
+        case EvaluationFailureErrorCode.ParseError:
+          throw new ParseError(details);
+        case EvaluationFailureErrorCode.TargetingKeyMissing:
+          throw new TargetingKeyMissingError(details);
+        case EvaluationFailureErrorCode.InvalidContext:
+          throw new InvalidContextError(details);
+        case EvaluationFailureErrorCode.FlagNotFound:
+          throw new FlagNotFoundError(details);
+        case EvaluationFailureErrorCode.General:
+          throw new TargetingKeyMissingError(details);
+        default:
+          throw new GeneralError(details);
+      }
+    }
+
+    if (typeof result.value.value !== typeof defaultValue) {
+      throw new TypeMismatchError();
+    }
+
+    return {
+      value: result.value.value as T,
+      variant: result.value.variant,
+      reason: result.value.reason,
+      flagMetadata: result.value.metadata && this.toFlagMetadata(result.value.metadata),
+    };
+  }
+
+  private toFlagMetadata(metadata: object): FlagMetadata {
+    // OFREP metadata is defined as any object but OF metadata is defined as Record<string, string | number | boolean>
+    const originalEntries = Object.entries(metadata);
+    const onlyPrimitiveEntries = originalEntries.filter(([, value]) =>
+      ['string', 'number', 'boolean'].includes(typeof value),
+    );
+    return Object.fromEntries(onlyPrimitiveEntries);
+  }
```

Comment:
@thomaspoignant we will have to see what should be done here.
Right now I am ignoring all non primitive values to make the `metadata` compatible with the real type and spec.
The problem here is that the OFREP type currently is object while it is stricter here in JS and the [spec](https://openfeature.dev/specification/types/#flag-metadata).

I see 3 possible solutions:
- Change the OFREP type for FlagMetadata to be the same as in the spec, and so also the same as the JS type
- Filter incompatible values (as I do here)
- Throw an error if the value coming from OFREP is not compatible with our solution

My preferred solution would be the first, as I think OFREP should not have broader `flagMetadata` than our specification as it creates this issue for no real value.

Go currently just does not provide a getter for a non primitive which is closest to the second option.
The FlagMetadata type there is `type FlagMetadata map[string]interface{}`, but there are only getters for primitives. 
https://github.com/open-feature/go-sdk/blob/2944608562d04521fcf92000948a22012e630471/openfeature/client.go#L185

Reply:
I would probably go for the 1st option.
Since OFREP is supposed to follow the OpenFeature spec, I see no reason to keep it less restrictive in OFREP.

Reply:
@lukas-reining sorry, I missed the notification on this discussion. 

I also agree with @thomaspoignant on this and thank you for fixing this already in the OFREP side :) I will check this for my go sdk implementation.

Reply:
Changed here https://github.com/open-feature/protocol/pull/8